src/foam/nanos/medusa/ClusterConfigMonitor.js

```diff
@@ -0,0 +1,210 @@
+
+            // // Wait for own replay to complete,
+            // // then set node ONLINE.
+            ReplayingInfo replaying = (ReplayingInfo) x.get("replayingInfo");
+            if ( replaying.getStartTime() == null ) {
+              replaying.setStartTime(new java.util.Date());
+            }
+            DAO dao = ((DAO) x.get("medusaNodeDAO"));
+            if ( replaying.getEndTime() == null ) {
+              replaying.setEndTime(new java.util.Date());
+              Max max = (Max) dao.select(MAX(MedusaEntry.INDEX));
+              replaying.setReplaying(false);
+              if ( max != null &&
+                   max.getValue() != null ) {
+                replaying.updateIndex(x, (Long)max.getValue());
+                replaying.setReplayIndex((Long)max.getValue());
+              }
+            }
+
+            // TODO: deal with digest failures - and Node taking self OFFLINE.
+            // this timer will continually set it back to ONLINE.
+
+            config = (ClusterConfig) config.fclone();
+            config.setStatus(Status.ONLINE);
+            ((DAO) x.get("localClusterConfigDAO")).put(config);
+
+             // TODO/REVIEW: disable monitor? is it needed
+          }
+        } else if ( config.getType() != MedusaType.MEDIATOR &&
+                    config.getType() != MedusaType.NERF &&
+                    config.getStatus() == Status.OFFLINE ) {
+          config = (ClusterConfig) config.fclone();
+          config.setStatus(Status.ONLINE);
+          ((DAO) x.get("localClusterConfigDAO")).put(config);
+        } else if ( support.getStandAlone() ) {
+          // standalone mode.
+          config = (ClusterConfig) config.fclone();
+          config.setStatus(Status.ONLINE);
+          config.setIsPrimary(true);
+          ((DAO) x.get("localClusterConfigDAO")).put(config);
+
+          DAO dao = (DAO) x.get("localClusterConfigDAO");
+          List<ClusterConfig> nodes = (ArrayList) ((ArraySink) dao.where(
+            AND(
+              EQ(ClusterConfig.ZONE, 0),
+              EQ(ClusterConfig.TYPE, MedusaType.NODE),
+              EQ(ClusterConfig.ENABLED, true),
+              EQ(ClusterConfig.STATUS, Status.OFFLINE)
+            ))
+          .select(new ArraySink())).getArray();
+          for ( ClusterConfig node : nodes ) {
+            node = (ClusterConfig) node.fclone();
+            node.setStatus(Status.ONLINE);
+            dao.put_(x, node);
+          }
+          // no need for timer in standalone mode.
+          Timer timer = (Timer) getTimer();
+          timer.cancel();
+          timer.purge();
+          getLogger().debug("timer", "cancel");
+          return;
+        }
+
+        // TODO: Non-Mediators just useful for reporting and network graph - the ping time could be reduced - see mn/services.jrl
+
+        // update ReplayingInfo for self.
+        DAO dao = (DAO) x.get("clusterConfigDAO");
+        dao.find_(x, config.getId());
```

Comment:
```suggestion
        DAO dao = (DAO) x.get("clusterConfigDAO");
        config = (ClusterConfig) dao.find_(x, config.getId());
```
Might not need to reload the config in case the id and realm were unchanging.

Reply:
See ClusterConfigReplayingInfoDAO - the find on self, is just to invoke this decorator which updates the config for network graph.